src/common/m_variables_conversion.fpp

```diff
@@ -128,39 +129,156 @@ end subroutine s_compute_abstract_average_state
+            call s_compute_pressure_4eqns(energy, alf, dyn_p, pi_inf, gamma, pres)
+        end if
+
+    end subroutine s_compute_pressure
+
+    !>  This subroutine is designed for the gamma/pi_inf model
+        !!      and provided a set of either conservative or primitive
+        !!      variables, transfers the density, specific heat ratio
+        !!      function and the liquid stiffness function from q_vf to
+        !!      rho, gamma and pi_inf.
+        !! @param q_vf conservative or primitive variables
         !! @param i cell index to transfer mixture variables
         !! @param j cell index to transfer mixture variables
         !! @param k cell index to transfer mixture variables
-        !! @param rho_K density
-        !! @param gamma_K  specific heat ratio function
-        !! @param pi_inf_K liquid stiffness
-        !! @param Re_k Reynolds number
-    subroutine s_convert_mixture_to_mixture_variables(qK_vf, rho_K, &
-                                                      gamma_K, pi_inf_K, &
-                                                      Re_K, i, j, k, G_K, G)
+        !! @param rho density
+        !! @param gamma  specific heat ratio function
+        !! @param pi_inf liquid stiffness
+    subroutine s_convert_mixture_to_mixture_variables(q_vf, i, j, k, &
+                                                      rho, gamma, pi_inf, Re_K, G_K, G)
 
-        type(scalar_field), dimension(sys_size), intent(IN) :: qK_vf
+        type(scalar_field), dimension(sys_size), intent(IN) :: q_vf
 
-        real(kind(0d0)), intent(OUT) :: rho_K, gamma_K, pi_inf_K
+        integer, intent(IN) :: i, j, k
 
-        real(kind(0d0)), dimension(2), intent(OUT) :: Re_K
+        real(kind(0d0)), intent(OUT), target :: rho
+        real(kind(0d0)), intent(OUT), target :: gamma
+        real(kind(0d0)), intent(OUT), target :: pi_inf
 
-        integer, intent(IN) :: i, j, k
+        real(kind(0d0)), optional, dimension(2), intent(OUT) :: Re_K
 
         real(kind(0d0)), optional, intent(OUT) :: G_K
         real(kind(0d0)), optional, dimension(num_fluids), intent(IN) :: G
 
-        ! Performing the transfer of the density, the specific heat ratio
-        ! function as well as the liquid stiffness function, respectively
-        rho_K = qK_vf(1)%sf(i, j, k)
-        gamma_K = qK_vf(gamma_idx)%sf(i, j, k)
-        pi_inf_K = qK_vf(pi_inf_idx)%sf(i, j, k)
+        real(kind(0d0)), pointer :: rho_K, gamma_K, pi_inf_K
+
+        !> Post process requires rho_sf/gamma_sf/pi_inf_sf to be 
+            !! updated instead of rho/gamma/pi_inf. Therefore, the
+            !! versions of these variables appended with '_K' represent
+            !! pointers that target the correct variable.
+#ifdef MFC_POST_PROCESS
+        rho_K => rho_sf(i, j, k)
+        gamma_K =>  gamma_sf(i, j, k)
+        pi_inf_K => pi_inf_sf(i, j, k)
+#endif
+
+#ifndef MFC_POST_PROCESS
+        rho_K  => rho
+        gamma_K => gamma
+        pi_inf_K => pi_inf
+#endif
```

Comment:
I think having a single `#else` directive instead of both `#endif` and `#ifndef` would make for more readable code. The `n` is quite hard to spot! There are a few places in this file where this is the case.